src/main/java/com/intellij/rt/debugger/agent/CaptureStorage.java

```diff
@@ -127,64 +225,74 @@ private static Method getGetCallerFrameMethod(Object key) throws NoSuchMethodExc
   }
 
   private static class ConcurrentIdentityWeakHashMap<K, V> {
-    private final ReferenceQueue referenceQueue = new ReferenceQueue();
-    private final ConcurrentMap<WeakReference, V> map = new ConcurrentHashMap<>();
```

Comment:
I was disappointed that the previous implementation relied on the fact that `HardKey` is a left-hand side argument of `equals`.